olive/cli/auto_opt.py

```diff
@@ -0,0 +1,265 @@
+
+    @staticmethod
+    def register_subcommand(parser: ArgumentParser):
+        sub_parser = parser.add_parser(
+            "auto-opt",
+            help=("Automatically performance optimize input model"),
+        )
+
+        add_hf_model_options(sub_parser)
+        add_logging_options(sub_parser)
+
+        # model options
+        model_attributes_group = sub_parser.add_argument_group("model attributes options")
+        model_attributes_group.add_argument(
+            "--model_framework",
+            type=str,
+            default="hf",
+            choices=["hf", "onnx", "pytorch"],
+            help="model type, choose from HfModel and ONNXModel",
+        )
+
+        system_group = sub_parser.add_argument_group("system options")
+        system_group.add_argument(
+            "--device",
+            type=str,
+            default=None,
+            choices=["cpu", "gpu", "npu"],
+            # TODO(anyone): add more devices cpu_spr, vpu, intel_myriad
+            help=(
+                "Device to use for optimization, choose from cpu and gpu. If not specified,"
+                " will deduce from the value of execution providers, CPU/VistisAi for cpu and"
+                " CUDA/Tensorrt/Dml for gpu. If both device and execution providers are not specified,"
+                " default to cpu device with CPUExecutionProvider."
+            ),
+        )
+        system_group.add_argument(
+            "--providers",
+            type=str,
+            nargs="*",
+            choices=["CPU", "CUDA", "Tensorrt", "Dml", "VitisAI", "Qnn"],
+            help="List of execution providers to use for optimization",
+        )
+
+        # dataset options
+        dataset_group = sub_parser.add_argument_group(
+            "dataset options, required for some optimization passes like quantization, and evaluation components"
+        )
+        dataset_group.add_argument(
+            "--data_config_path",
+            type=str,
+            help="Path to the data config file. It allows to customize the data config(json/yaml) for the model.",
+        )
+
+        auto_opt_config_group = sub_parser.add_argument_group("auto optimizer options")
+        auto_opt_config_group.add_argument(
+            "--precisions",
+            type=str,
+            nargs="*",
+            choices=["fp16", "fp32", "int4", "int8"],
+            help=(
+                "The output precision of the optimized model. If not specified, "
+                "the default precision is fp32 for cpu and fp16 for gpu"
+            ),
+        )
+        auto_opt_config_group.add_argument(
+            "--excluded_passes",
+            type=str,
+            nargs="*",
+            help=(
+                "List of passes to disable for optimization, if not specified, "
+                "auto-opt will disable ModelBuilder/OrtPerfTuning by default."
+            ),
+        )
+        auto_opt_config_group.add_argument(
+            "--use_model_builder",
```

Comment:
Can we abstract MB decision point for user? As a first-time user, this will add cognitive load ("what is model builder; should I use this"). Instead, could we test whether MB supports the model and if it does just use MB?

![image](https://github.com/user-attachments/assets/8dbf9835-fe58-4390-b476-25c8287404c8)

Reply:
We need an API from MB to check whether the model is supported or not. We can do this as a follow up PR.